Update isValueOutOfRangeForProperty() to account for more CSS properties

cdumez opened the pull request:
#### e864f3817fb82d58b9a9c874ab45e35c8c40e0ae
<pre>
Update isValueOutOfRangeForProperty() to account for more CSS properties
<a href="https://bugs.webkit.org/show_bug.cgi?id=249050">https://bugs.webkit.org/show_bug.cgi?id=249050</a>

Reviewed by Sam Weinig.

Update isValueOutOfRangeForProperty() to account for more CSS properties that
have range limits:
- border-image-width: length or percentage or number &gt;= 0
  <a href="https://w3c.github.io/csswg-drafts/css-backgrounds/#border-image-width">https://w3c.github.io/csswg-drafts/css-backgrounds/#border-image-width</a>

- border-top-width, border-right-width, border-bottom-width, border-left-width: length &gt;= 0
  <a href="https://w3c.github.io/csswg-drafts/css-backgrounds/#the-border-width">https://w3c.github.io/csswg-drafts/css-backgrounds/#the-border-width</a>

- row-gap, column-gap: length or percentage &gt;= 0
  <a href="https://w3c.github.io/csswg-drafts/css-align/#column-row-gap">https://w3c.github.io/csswg-drafts/css-align/#column-row-gap</a>

- column-width: length &gt;= 0
  <a href="https://w3c.github.io/csswg-drafts/css-multicol-1/#propdef-column-width">https://w3c.github.io/csswg-drafts/css-multicol-1/#propdef-column-width</a>

- flex-basis: length or percentage &gt;= 0
  <a href="https://w3c.github.io/csswg-drafts/css-flexbox/#flex-basis-property">https://w3c.github.io/csswg-drafts/css-flexbox/#flex-basis-property</a>

- grid-auto-columns, grid-auto-rows: length or percentage &gt;= 0
  <a href="https://w3c.github.io/csswg-drafts/css-grid/#auto-tracks">https://w3c.github.io/csswg-drafts/css-grid/#auto-tracks</a>

- line-height: length, number or percentage &gt;= 0
  <a href="https://w3c.github.io/csswg-drafts/css-inline/#line-height-property">https://w3c.github.io/csswg-drafts/css-inline/#line-height-property</a>

- max-height: length or percentage &gt;= 0
  <a href="https://w3c.github.io/csswg-drafts/css2/#propdef-max-height">https://w3c.github.io/csswg-drafts/css2/#propdef-max-height</a>

- max-width: length or percentage &gt;= 0
  <a href="https://w3c.github.io/csswg-drafts/css2/#propdef-max-width">https://w3c.github.io/csswg-drafts/css2/#propdef-max-width</a>

- outline-width: length &gt;= 0
  <a href="https://w3c.github.io/csswg-drafts/css-ui/#outline-width">https://w3c.github.io/csswg-drafts/css-ui/#outline-width</a>

- scroll-padding-inline-start, scroll-padding-block-start, scroll-padding-inline-end, scroll-padding-block-end: length or percentage &gt;= 0
  <a href="https://w3c.github.io/csswg-drafts/css-scroll-snap/#padding-longhands-logical">https://w3c.github.io/csswg-drafts/css-scroll-snap/#padding-longhands-logical</a>

- stroke-miterlimit: number &gt;= 0
  <a href="https://svgwg.org/svg2-draft/painting.html#StrokeMiterlimitProperty">https://svgwg.org/svg2-draft/painting.html#StrokeMiterlimitProperty</a>

- font-weight: number in range [1, 1000]
  <a href="https://w3c.github.io/csswg-drafts/css-fonts/#font-weight-prop">https://w3c.github.io/csswg-drafts/css-fonts/#font-weight-prop</a>

All these properties are covered by the WPT tests but the tests still fail because of:
- <a href="https://github.com/WebKit/WebKit/pull/7387">https://github.com/WebKit/WebKit/pull/7387</a> (a lot more tests start passing with this PR)

For now, I have extracted these test cases into their own layout test.

* LayoutTests/fast/css/css-typed-om/style-property-map-set-negative-value-expected.txt:
* LayoutTests/fast/css/css-typed-om/style-property-map-set-negative-value.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/animation-duration-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/animation-iteration-count-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/border-width-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/column-width-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/flex-basis-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/font-weight-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/gap-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/grid-auto-columns-rows-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/height-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/line-height-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/logical-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/outline-width-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/scroll-padding-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/stroke-miterlimit-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/width-expected.txt:
* Source/WebCore/css/typedom/CSSUnitValue.cpp:
(WebCore::isValueOutOfRangeForProperty):

Canonical link: <a href="https://commits.webkit.org/257757@main">https://commits.webkit.org/257757@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e41fe3c534b35716555a53a53ea18762b23a22bb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/99911 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/9083 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/32993 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/109264 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/169502 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/103913 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/9964 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/86377 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/92361 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/107167 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/105678 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/7520 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/90805 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/34246 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/89426 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/22196 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/2884 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/23709 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/2840 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/46081 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/8973 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/43182 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2733 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/4692 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

weinig reviewed (approved):
Talked with Chris. Going to need to play with the generator to get this right, so lets land this and I will take a crack at the generation.

cdumez committed:
Update isValueOutOfRangeForProperty() to account for more CSS properties
https://bugs.webkit.org/show_bug.cgi?id=249050

Reviewed by Sam Weinig.

Update isValueOutOfRangeForProperty() to account for more CSS properties that
have range limits:
- border-image-width: length or percentage or number >= 0
  https://w3c.github.io/csswg-drafts/css-backgrounds/#border-image-width

- border-top-width, border-right-width, border-bottom-width, border-left-width: length >= 0
  https://w3c.github.io/csswg-drafts/css-backgrounds/#the-border-width

- row-gap, column-gap: length or percentage >= 0
  https://w3c.github.io/csswg-drafts/css-align/#column-row-gap

- column-width: length >= 0
  https://w3c.github.io/csswg-drafts/css-multicol-1/#propdef-column-width

- flex-basis: length or percentage >= 0
  https://w3c.github.io/csswg-drafts/css-flexbox/#flex-basis-property

- grid-auto-columns, grid-auto-rows: length or percentage >= 0
  https://w3c.github.io/csswg-drafts/css-grid/#auto-tracks

- line-height: length, number or percentage >= 0
  https://w3c.github.io/csswg-drafts/css-inline/#line-height-property

- max-height: length or percentage >= 0
  https://w3c.github.io/csswg-drafts/css2/#propdef-max-height

- max-width: length or percentage >= 0
  https://w3c.github.io/csswg-drafts/css2/#propdef-max-width

- outline-width: length >= 0
  https://w3c.github.io/csswg-drafts/css-ui/#outline-width

- scroll-padding-inline-start, scroll-padding-block-start, scroll-padding-inline-end, scroll-padding-block-end: length or percentage >= 0
  https://w3c.github.io/csswg-drafts/css-scroll-snap/#padding-longhands-logical

- stroke-miterlimit: number >= 0
  https://svgwg.org/svg2-draft/painting.html#StrokeMiterlimitProperty

- font-weight: number in range [1, 1000]
  https://w3c.github.io/csswg-drafts/css-fonts/#font-weight-prop

All these properties are covered by the WPT tests but the tests still fail because of:
- https://github.com/WebKit/WebKit/pull/7387 (a lot more tests start passing with this PR)

For now, I have extracted these test cases into their own layout test.

* LayoutTests/fast/css/css-typed-om/style-property-map-set-negative-value-expected.txt:
* LayoutTests/fast/css/css-typed-om/style-property-map-set-negative-value.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/animation-duration-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/animation-iteration-count-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/border-width-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/column-width-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/flex-basis-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/font-weight-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/gap-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/grid-auto-columns-rows-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/height-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/line-height-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/logical-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/outline-width-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/scroll-padding-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/stroke-miterlimit-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/width-expected.txt:
* Source/WebCore/css/typedom/CSSUnitValue.cpp:
(WebCore::isValueOutOfRangeForProperty):

Canonical link: https://commits.webkit.org/257757@main
